Content.Server/Fluids/EntitySystems/SpraySystem.cs

```diff
@@ -57,42 +57,41 @@ private void OnAfterInteract(EntityUid uid, SprayComponent component, AfterInter
             return;
         }
 
-        var playerPos = Transform(args.User).Coordinates;
+        var playerMapPos = Transform(args.User).MapPosition;
+        var clickMapPos = args.ClickLocation.ToMap(EntityManager);
 
-        if (args.ClickLocation.GetGridUid(EntityManager) != playerPos.GetGridUid(EntityManager))
+        var diffPos = clickMapPos.Position - playerMapPos.Position;
+        var diffLength = diffPos.Length;
+        if (diffLength == 0f)
             return;
 
-        var direction = (args.ClickLocation.Position - playerPos.Position).Normalized;
-        var threeQuarters = direction * 0.75f;
-        var quarter = direction * 0.25f;
+        var diffNorm = diffPos.Normalized;
+        var threeQuarters = diffNorm * 0.75f;
+        var quarter = diffNorm * 0.25f;
 
         var amount = Math.Max(Math.Min((solution.CurrentVolume / component.TransferAmount).Int(), component.VaporAmount), 1);
 
         var spread = component.VaporSpread / amount;
 
         for (var i = 0; i < amount; i++)
         {
-            var rotation = new Angle(direction.ToAngle() + Angle.FromDegrees(spread * i) -
+            var rotation = new Angle(diffNorm.ToAngle() + Angle.FromDegrees(spread * i) -
```

Comment:
diffNorm should just be a local variable outside of the loop as the angle is being called multiple times.